src/lib/cfa/ntp_conf.rb

```diff
@@ -0,0 +1,248 @@
+require "cfa/base_model"
+require "cfa/augeas_parser"
+require "cfa/matcher"
+
+module CFA
+  # class representings /etc/ntp.conf file model. It provides helper to manipulate
+  # with file. It uses CFA framework and Augeas parser.
+  # @see http://www.rubydoc.info/github/config-files-api/config_files_api/CFA/BaseModel
+  # @see http://www.rubydoc.info/github/config-files-api/config_files_api/CFA/AugeasParser
+  class NtpConf < ::CFA::BaseModel
+    PARSER = CFA::AugeasParser.new("ntp.lns")
+    DEFAULT_PATH = "/etc/ntp.conf".freeze
+    SIMPLE_SETTINGS = ["logfile", "driftfile"].freeze
+    AUTH_COMMANDS = ["trustedkey", "requestkey", "controlkey", "keys"].freeze
+    COMMAND_RECORDS = [
+      "server",
+      "peer",
+      "broadcast",
+      "manycastclient",
+      "multicastclient",
+      "manycastserver"
+    ].freeze
+
+    def self.ntp_attributes(*attrs)
+      attrs.each do |attr|
+        define_method(attr) { get_attribute(attr.to_s) }
+        define_method("#{attr}=") { |record| set_attribute(attr.to_s, record) }
+        define_method("delete_#{attr}") { delete_attribute(attr.to_s) }
+      end
+    end
+
+    ntp_attributes :driftfile, :logfile, :keys, :requestkey, :controlkey
+
+    def initialize(path: DEFAULT_PATH, file_handler: nil)
+      super(PARSER, path, file_handler: file_handler)
+    end
+
+    def servers
+      RecordCollection.new(ServerRecord, data.collection("server"))
+    end
+
+    def peers
+      RecordCollection.new(PeerRecord, data.collection("peer"))
+    end
+
+    def restricts
+      RecordCollection.new(RestrictRecord, data.collection("restrict"))
+    end
+
+    private
+
+    def get_attribute(name)
+      return nil if data[name].nil?
+      AttributeRecord.new_from_augeas(data[name])
+    end
+
+    def set_attribute(name, record)
+      data[name] = record.to_augeas
+    end
+
+    def delete_attribute(name)
+      data.delete(name)
+    end
+
+    # class to manage a collection of some ntp entries, as
+    # server, restrict, peer, etc.
+    #   For example:
+    #     server 0.opensuse.pool.ntp.org iburst
+    #     server 1.opensuse.pool.ntp.org iburst
+    #     server 2.opensuse.pool.ntp.org iburst
+    #     server 3.opensuse.pool.ntp.org iburst
+    class RecordCollection
+      include Enumerable
+
+      def initialize(record_class, augeas_collection)
+        @record_class = record_class
+        @augeas_collection = augeas_collection
+      end
+
+      def each
+        @augeas_collection.each do |augeas_value|
+          yield @record_class.new_from_augeas(augeas_value)
+        end
+      end
+
+      def add(record)
+        @augeas_collection.add(record.to_augeas)
+      end
+
+      def delete(record)
+        @augeas_collection.delete(record.to_augeas)
+      end
+
+      def replace(old_record, new_record)
+        return if !include? old_record
+        matcher = CFA::Matcher.new { |_k, v| v == old_record.to_augeas }
```

Comment:
wrong matcher, I think better to use `matcher = CFA::Matcher.new(value_matcher: old_record.to_augeas)`